x-pack/test/functional_gen_ai/inference/tests/chat_complete.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { lastValueFrom, toArray } from 'rxjs';
+import expect from '@kbn/expect';
+import { supertestToObservable } from '@kbn/sse-utils-server';
+import type { AvailableConnectorWithId } from '@kbn/gen-ai-functional-testing';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+export const chatCompleteSuite = (
```

Comment:
Do you think it would be worthwhile to test a failure scenario?
e.g. https://github.com/pgayvallet/kibana/blob/afa9b8a540061a0f00ebf20b80fe43ebf562a372/x-pack/plugins/inference/server/chat_complete/api.ts#L54-L58

Reply:
It was a little tricky because we're calling the API through http, and error transport requires transformation, but I added it in https://github.com/elastic/kibana/pull/198000/commits/2efbf5a082294226b284cb08af8cff0e2c01f3f2

Reply:
@pgayvallet nice!